src/content/reducers/url-state.js

```diff
@@ -151,6 +166,44 @@ function hidePlatformDetails(state: boolean = false, action: Action) {
   }
 }
 
+function threadOrder(state: ThreadIndex[] = [], action: Action) {
+  switch (action.type) {
+    case 'RECEIVE_PROFILE_FROM_ADDON':
+    case 'RECEIVE_PROFILE_FROM_WEB':
+    case 'RECEIVE_PROFILE_FROM_FILE': {
+      // When receiving a new profile, try to use the thread order specified in the URL,
+      // but ensure that the IDs are correct.
+      const threads = defaultThreadOrder(action.profile.threads);
+      const validURLThreads = state.filter(index => threads.includes(index));
+      const missingThreads = threads.filter(index => !state.includes(index));
+      return validURLThreads.concat(missingThreads);
+    }
+    case 'CHANGE_THREAD_ORDER': {
+      const { threadOrder } = action;
+      return threadOrder;
+    }
+    default:
+      return state;
+  }
+}
+
+function hiddenThreads(state: ThreadIndex[] = [], action: Action) {
+  switch (action.type) {
+    case 'HIDE_THREAD': {
+      const { threadIndex, threadOrder } = action;
+      // Do not allow hiding the last thread.
+      const isLastVisibleThread = state.length + 1 === threadOrder.length;
```

Comment:
Also, I'd move the condition to the dispatcher in https://github.com/devtools-html/perf.html/pull/380/files#diff-694fe6b070eafcb88831ddf4e0c7a5b3R40

By replacing the simple dispatcher by a slightly more complex:
```javascript
export function hideThread(
  threadIndex: ThreadIndex,
  threadOrder: ThreadIndex[],
  hiddenThreads: ThreadIndex[]
): ThunkAction {
  return dispatch => {
    if (threadOrder.length - state.length <=1) {
      return;
    }

    return dispatch({
      type: 'HIDE_THREAD',
      threadIndex,
      threadOrder,
      hiddenThreads,
    });
  };
}
```

This way you don't need the comparison in the reducers anymore. Maybe you won't need the `threadOrder` in the action anymore too :)

Bonus points if you can "disable" the checkbox in the menu item in that case, but I don't know if that's possible.